src/Fls.Results/OperationResult.cs

```diff
@@ -81,21 +152,36 @@ public FailureResult(Exception exception)
             }
         }
 
+        /// <summary>
+        /// Produces success value type for given value
+        /// </summary>
         public static IOperationResult<T> Success<T>(T value)
         {
             return new SuccessResult<T>(value);
         }
 
+        /// <summary>
+        /// Produces error value type for given error message text
+        /// </summary>
         public static IOperationResult<T> Error<T>(string message)
         {
             return new ErrorResult<T>(message);
         }
 
+        /// <summary>
+        /// Produces failure value type for given exception
+        /// </summary>
         public static IOperationResult<T> Failure<T>(Exception exception)
         {
             return new FailureResult<T>(exception);
         }
-
+    
+        /// <summary>
+        /// Binds the result of previous operation to suitable value
```

Comment:
Suggested change (we need these comments as helpful as they can get. imagine us trying to figure this thing out a month from now):

If <code></code> matches as a success it is bound using <code>bind</code>, otherwise it is passed on.